dvc/repo/ls.py

```diff
@@ -83,24 +83,24 @@ def _ls(
     fs_path = fs.info(fs_path)["name"]
 
     infos = {}
-    for root, dirs, files in fs.walk(
-        fs_path, dvcfiles=True, dvc_only=dvc_only, detail=True
-    ):
-        if not recursive:
-            files.update(dirs)
-
-        parts = fs.relparts(root, fs_path)
-        if parts == (".",):
-            parts = ()
+    if fs.isfile(fs_path):
+        infos[os.path.basename(path)] = fs.info(fs_path)
+    else:
```

Comment:
I have switched the `if-else` here, since `fs.walk()` also returns result for files now.